app/validators/redirect_uri_validator.rb

```diff
@@ -29,14 +26,24 @@ def validate_each(record, attribute, value)
 
   private
 
-  def native_redirect_uri?(uri)
-    self.class.native_redirect_uri.present? && uri.to_s == self.class.native_redirect_uri.to_s
+  def oob_redirect_uri?(uri)
+    Doorkeeper::OAuth::NonStandard::IETF_WG_OAUTH2_OOB_METHODS.include?(uri)
   end
 
   def forbidden_uri?(uri)
     Doorkeeper.configuration.forbid_redirect_uri.call(uri)
   end
 
+  def unspecified_scheme?(uri)
+    return true unless uri.opaque.nil?
```

Comment:
I personally prefer `if uri.opaque` (or `opaque.present?`) for simplicity

Reply:
ok. easy improvement. done.